lib/internal/Magento/Framework/App/Test/Unit/ProductMetadataTest.php

```diff
@@ -20,16 +21,27 @@ class ProductMetadataTest extends \PHPUnit\Framework\TestCase
      */
     private $composerInformationMock;
 
+    /**
+     * @var CacheInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $cacheMock;
+
     protected function setUp()
     {
         $this->composerInformationMock = $this->getMockBuilder(\Magento\Framework\Composer\ComposerInformation::class)
             ->disableOriginalConstructor()->getMock();
 
+        $this->cacheMock = $this->getMockBuilder(CacheInterface::class)->getMock();
+
         $objectManager = new ObjectManager($this);
         $this->productMetadata = $objectManager->getObject(ProductMetadata::class);
         $reflectionProperty = new \ReflectionProperty($this->productMetadata, 'composerInformation');
         $reflectionProperty->setAccessible(true);
         $reflectionProperty->setValue($this->productMetadata, $this->composerInformationMock);
+
+        $reflectionProperty = new \ReflectionProperty($this->productMetadata, 'cache');
```

Comment:
`cache` property could be passed to constructor, no need to add it via reflection

Following should work:
```php
$this->cacheMock = $this->getMockBuilder(CacheInterface::class)->getMock();

$objectManager = new ObjectManager($this);
$this->productMetadata = $objectManager->getObject(ProductMetadata::class, ['cache' => $this->cacheMock]);
```

Reply:
Updated. Please check now.